builtin/nomad/auth.go

```diff
@@ -0,0 +1,11 @@
+package nomad
+
+import "os"
+
+func ConsulAuth() (string, error) {
+	return os.Getenv("CONSUL_HTTP_TOKEN"), nil
+}
+
+func VaultAuth() (string, error) {
+	return os.Getenv("VAULT_TOKEN"), nil
```

Comment:
It would be good to confirm this works with remote runners if you haven't already given that a shot with this PR. I think you'd have to do a `waypoint config set -runner CONSUL_HTTP_TOKEN=abcd VAULT_TOKEN=1234` and then try an Up operation with ODR/Remote runners and see that the deployment succeeded with these values.

Reply:
@briancain Looking good! I first unset the tokens to verify that it would fail without them w/remote-operations:

```terminal
% waypoint config set -scope=global -runner VAULT_TOKEN=                        

% waypoint config set -scope=global -runner CONSUL_HTTP_TOKEN=

% waypoint deploy -p=example-nodejs -local=false -release=false

» Deploying example-nodejs...
  Performing this operation on a remote runner with id "JOB_ID"

» Cloning data from Git
  URL: https://github.com/hashicorp/waypoint-examples
  Ref: feat/nomad-consul-vault-integration
✓ Running deploy v14
✓ Deleting job: 
! 2 errors occurred:
  	* Unexpected response code: 500 (Vault policies requested but missing Vault
  Token)
  	* Error during rollback: Unexpected response code: 500 (missing job ID for
  deregistering)
```

After setting them again, the job deployed:
```terminal
% waypoint config set -scope=global -runner CONSUL_HTTP_TOKEN=my-token

% waypoint config set -scope=global -runner VAULT_TOKEN=my-token

% waypoint deploy -p=example-nodejs -local=false -release=false                                   

» Deploying example-nodejs...
  Performing this operation on a remote runner with id "JOB_ID"

» Cloning data from Git
  URL: https://github.com/hashicorp/waypoint-examples
  Ref: feat/nomad-consul-vault-integration
✓ Job registration successful
✓ Allocation "ALLOC_ID" created: node "NODE_ID", group "app"
✓ Evaluation status changed: "pending" -> "complete"
✓ Evaluation "EVAL_ID" finished with status "complete"
✓ Deployment successfully rolled out!
✓ Running deploy v15

  Performing this operation on a remote runner with id "JOB_ID"

» Cloning data from Git
  URL: https://github.com/hashicorp/waypoint-examples
  Ref: feat/nomad-consul-vault-integration
✓ Finished building report for Nomad platform
✓ Getting job info...
✓ Job "web" is reporting ready!
```

Reply:
Nice! 🎉  Looks great